internal/services/oracledatabase/exadata_infrastructure_resource_test.go

```diff
@@ -0,0 +1,214 @@
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ExadataInfraResource struct{}
+
+func (a ExadataInfraResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := cloudexadatainfrastructures.ParseCloudExadataInfrastructureID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.OracleDatabase.OracleDatabaseClient.CloudExadataInfrastructures.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Exadata Infrastructure %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (a ExadataInfraResource) Destroy(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := cloudexadatainfrastructures.ParseCloudExadataInfrastructureID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := client.OracleDatabase.OracleDatabaseClient.CloudExadataInfrastructures.Delete(ctx, *id); err != nil {
+		return nil, fmt.Errorf("deleting Exadata Infrastructure %s: %+v", id, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func TestExaInfra_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestExaInfra_allFields(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.allFields(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestExaInfra_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestExaInfra_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, oracledatabase.ExadataInfraResource{}.ResourceType(), "test")
+	r := ExadataInfraResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (a ExadataInfraResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+
+
+%s
```

Comment:
can we tidy this up??

```suggestion
	return fmt.Sprintf(`
%s
```